### PR TITLE
Initial work to support ieee802.1x (enterprise wifi)

### PR DESCRIPTION
**Pull request by Herrie82**

### CODE REVIEW

File: src/wifi_service.c [hunk 1]
```diff
@@ -943,7 +958,7 @@ None
  *  luna://com.palm.wifi/connect '{"ssid":"<Name of the access point>",
  *                                 "security": { "securityType": "",
  *                                     "simpleSecurity": { "passKey": "<passphrase for the network>" },
- *                                     "enterpriseSecurity": { ... }
+ *                                     "enterpriseSecurity": { "idenTity": "<username for the network", "passKey": "<passphrase for the network>"  }
```
Comment (Tofee) [inline src/wifi_service.c:961]: uppercase t, a typo maybe ?

**Comment by Herrie82** [comment]
It's a comment, will fix it after merge. Since PR is from elvispre's branch I cannot fix it directly.

File: src/wifi_service.c [hunk 2]
```diff
@@ -625,10 +636,14 @@ static void connect_wifi_with_ssid(const char *ssid, jvalue_ref req_object, luna
...
+		else if (jobject_get_exists(security_obj, J_CSTR_TO_BUF("enterpriseSecurity"), &enterprise_security_obj) &&
+			 jobject_get_exists(enterprise_security_obj, J_CSTR_TO_BUF("identityEAP"), &identity_obj) &&
+			 jobject_get_exists(enterprise_security_obj, J_CSTR_TO_BUF("passKey"), &passkey_obj))
 		{
-			LSMessageReplyCustomError(service_req->handle, service_req->message, "Not implemented");
-			goto cleanup;
+			identity_buf = jstring_get(identity_obj);
+			settings->identity = strdup(identity_buf.m_str);			
+			passkey_buf = jstring_get(passkey_obj);
+			settings->passkey = strdup(passkey_buf.m_str);
```
Comment (nizovn) [inline src/wifi_service.c:646]: It's better to free used raw_buffers using jstring_free_buffer(), as there is a small memory leak.

Comment (Herrie82) [inline src/wifi_service.c:646]: Like this? https://github.com/Herrie82/webos-connman-adapter/commit/40c295d3150faa0fe45e958ab153330a549ce9e5

Comment (nizovn) [inline src/wifi_service.c:646]: yes :)

Comment (Herrie82) [inline src/wifi_service.c:646]: OK that's in now :)

File: src/wifi_service.c (hunk 1 above)
Comment (Herrie82) [inline src/wifi_service.c:961]: Typo fixed as well now :)

**Pull request merged**